Wchan29/Add critical speed analyzer

wchan29 opened the pull request:
# Issue
Address issue https://github.com/Severson-Group/eMach/issues/298

# Summary
This PR adds rotor critical speed analyzer to eMach codebase.

closes #298

wchan29 committed:
init push

wchan29 committed:
add rst

wchan29 committed:
add example output to rst

wchan29 committed:
add result class to analyzer

dmnewman3 committed:
Merge branch 'develop' into wchan29/add_critical_speed_analyzer

dmnewman3 committed:
Merge branch 'wchan29/add_critical_speed_analyzer' of https://github.com/Severson-Group/eMach into wchan29/add_critical_speed_analyzer

dmnewman3 committed:
merge with develop and clean up grammar of .rst file

dmnewman3 reviewed (changes requested):
@wchan29 Here is my Level 1 review summary:

- Does the code run without error and produce the expected result? **Yes**
- Does the code comply with the [code guidelines](https://emach.readthedocs.io/en/latest/code.html)? **Yes**
- Does the code documentation comply with the [documentation guidelines](https://emach.readthedocs.io/en/latest/documentation.html)? **No, see below** 
- Is the writing, grammar, and syntax correct and clear? **Yes**
- Is the changeset compliant with the eMach architecture? **Yes**
- Does this review consider whether this physics are accurate? **No**
- Is PR approved to Level 2? **No**

Please include the image you referenced in the script of the .rst file instead of just referencing it. The way to do this is to place the .svg images in the \images folder of the docs and create an image block similar to other mechanical and EM analyzers.

wchan29 committed:
add bc table into rst

wchan29 committed:
add bc table into rst

wchan29 commented:
@dmnewman3 I have added the figure as requested. Please re-review.

dmnewman3 reviewed (dismissed):
Well done @wchan29! Here is my second Level 1 review summary:

- Does the code run without error and produce the expected result? **Yes**
- Does the code comply with the [code guidelines](https://emach.readthedocs.io/en/latest/code.html)? **Yes**
- Does the code documentation comply with the [documentation guidelines](https://emach.readthedocs.io/en/latest/documentation.html)? **Yes** 
- Is the writing, grammar, and syntax correct and clear? **Yes**
- Is the changeset compliant with the eMach architecture? **Yes**
- Does this review consider whether this physics are accurate? **No**
- Is PR approved to Level 2? **Yes**

This is approved to Level 2 @elsevers @npetersen2 @Anvar-Khamitov

wchan29 commented:
@elsevers @Anvar-Khamitov @npetersen2 This is ready for Level 2 review.

elsevers committed:
edit language

elsevers reviewed (changes requested):
# Level 2 review summary:
*Nice, organized, professional PR here. This is close to being ready to merge. I have just a couple requests, one of which I am worried may take a bit of effort to accomplish. Details below.*

- Does the code comply with the [code guidelines](https://emach.readthedocs.io/en/latest/code.html)? **Yes**
- Does the code documentation comply with the [documentation guidelines](https://emach.readthedocs.io/en/latest/documentation.html)? **Yes**
- Is the writing, grammar, and syntax correct and clear? **Yes** (I made some edits)
- Is the changeset compliant with the eMach architecture? **Yes**
- Are the physics accurate? **I did not verify**
- Level 1 re-review instructions (if revisions are requested): **Please confirm that replacement to Fig. 1 is clear on how one should select $\beta_{fi}$.**

# Changes requested
1. **The big one:** Figure 1 is a screenshot from a publication. And while it is very good that you cite the source (this avoids plagiarism), copy-pasting an image like this creates a copyright issue. I think we code this into our RST file natively (i.e., no screenshot, instead create a table using RST code; probably use Inkscape to make the images that go with it). 
2. Figure 1 is providing values of $\beta_n l$. It is not clear how these correspond to $\beta_{fi}$ in the equation above it?
3. The table `Input for rotor critical speed problem` is missing $\beta_{fi}$

wchan29 committed:
modified code block comment

wchan29 committed:
change beta_fi to betaL

wchan29 committed:
add .svg for BC constants and modify beta_fi in code to beta_l

wchan29 committed:
add message to clarify beta

wchan29 commented:
I have made changes as requested by @elsevers 

> 1. The big one: Figure 1 is a screenshot from a publication. And while it is very good that you cite the source (this avoids plagiarism), copy-pasting an image like this creates a copyright issue. I think we code this into our RST file natively (i.e., no screenshot, instead create a table using RST code; probably use Inkscape to make the images that go with it).

I was able to create a table in Inkscape similar to the one from the textbook, I simplified it so only the diagrams and constant values are in the table.

> 2. Figure 1 is providing values of $\beta_n l$. It is not clear how these correspond to $\beta_{fi}$ in the equation above it?

I modified the input for the problem class from `beta_fi` to `beta_l` so everything should be consistent. 

> 3. The table Input for rotor critical speed problem is missing $\beta_{fi}$.

Should be fixed now. 

@dmnewman3 ready for Level 1 re-review.

dmnewman3 reviewed (dismissed):
@wchan29 The changes look good and everything compiles properly. Here is my re-review:

- Does the code run without error and produce the expected result? **Yes**
- Does the code comply with the [code guidelines](https://emach.readthedocs.io/en/latest/code.html)? **Yes**
- Does the code documentation comply with the [documentation guidelines](https://emach.readthedocs.io/en/latest/documentation.html)? **Yes** 
- Is the writing, grammar, and syntax correct and clear? **Yes**
- Is the changeset compliant with the eMach architecture? **Yes**
- Does this review consider whether this physics are accurate? **No**
- Is PR approved to Level 2? **Yes**

@elsevers Ball is in your court again

wchan29 committed:
change beta_fi to beta_l in main code

wchan29 commented:
Ready for 2nd level review again. @elsevers @npetersen2 @Anvar-Khamitov

wchan29 commented:
Bumping this again, Ready for 2nd level review again. @elsevers @npetersen2 @Anvar-Khamitov

elsevers committed:
edit language in output section

elsevers committed:
edit language in output section

elsevers reviewed (approved):
Level 2 review summary: **Nice job, I approve merging this in**

- Does the code comply with the [code guidelines](https://emach.readthedocs.io/en/latest/code.html)? **Yes**
- Does the code documentation comply with the [documentation guidelines](https://emach.readthedocs.io/en/latest/documentation.html)? **Yes**
- Is the writing, grammar, and syntax correct and clear? **Yes** (note, I did make some edits in the last section of the RST file)
- Is the changeset compliant with the eMach architecture? **Yes**
- Are the physics accurate? **I did not check this**
- Did the reviewer change [the release schedule](https://github.com/orgs/Severson-Group/projects/26/views/1) for the issues bing closed? **Yes**, I scheduled this for `v1.1.0` release because it adds a feature.
- Did the reviewer change the branch the PR is being merged into? **No**, this is configured to merge into `develop`, which is correct.
- Level 1 re-review instructions (if revisions are requested):
- When your PR is approved, **remember to merge via `Squash and Merge`**. Please ask if you are unsure of how to do this.

elsevers commented:
@dmnewman3 , can you please `approve` this PR? (your comments indicate you approve, my request is just to make the green checkbox appear next to your name)

dmnewman3 commented:
> @dmnewman3 , can you please `approve` this PR? (your comments indicate you approve, my request is just to make the green checkbox appear next to your name)

All set @elsevers. All set to merge @wchan29.
